pkg/webhook/webhook.go

```diff
@@ -259,6 +259,65 @@ func writeResponse(w http.ResponseWriter, ar *v1beta1.AdmissionReview) {
 	w.Write(resp)
 }
 
+func addVolDownwardAPI(patch []jsonPatchOperation) []jsonPatchOperation {
+	labels := corev1.ObjectFieldSelector{
+		FieldPath:"metadata.labels",
+	}
+	dAPILabels := corev1.DownwardAPIVolumeFile{
+		Path: "labels",
+		FieldRef: &labels,
+	}
+	annotations := corev1.ObjectFieldSelector{
+		FieldPath:"metadata.annotations",
+	}
+	dAPIAnnotations := corev1.DownwardAPIVolumeFile{
+		Path: "annotations",
+		FieldRef: &annotations,
+	}
+	dAPIItems := []corev1.DownwardAPIVolumeFile{dAPILabels, dAPIAnnotations}
+	dAPIVolSource := corev1.DownwardAPIVolumeSource{
+		Items: dAPIItems,
+	}
+	volSource := corev1.VolumeSource{
+		DownwardAPI: &dAPIVolSource,
+	}
+	vol := corev1.Volume{
+		Name: "podinfo",
```

Comment:
I think `/etc/podinfo` path is from downward API example, we can decide on a unique path for pod network information. @przemeklal  would like to hear your thoughts on how to fix it? I don't think we could use a random path here because application will need to know where to find the downward API files.

Reply:
Yup, that's why I suggested we call it a limitation and mention this in the readme 😉
Agree that some random path is even worse - the only thing that comes to my mind at the moment is that we have access to the full pod spec - maybe first we could see if the "podinfo" volume exists, what's its type and then try patch it with our annotations/labels, however this seems like an over-complication.

We could also decide to overwrite it, don't do anything at all if some other volume with the same name is already there or maybe don't allow the pod to be created by sending appropriate admission review response result with the explanation.

Personally I'd be fine with leaving it as it is - at this stage we don't even handle preexisting resources requests neither - just overwrite them.

Reply:
Thanks for the input and comments. Seems like there is agreement to leave this as is, and just be sure to add it to the readme. I will see if I can figure out a way to deal with this gracefully if the podinfo volume exists, in a simple way.

Reply:
As Zenghui stated, /etc/podinfo is from the downward API doc example. I will change the path to /etc/podnetinfo to hopefully reduce the chances of that path already existing (from others that may copy downward API doc example).